iothub/service/src/DigitalTwin/DigitalTwinsClient.cs

```diff
@@ -0,0 +1,363 @@
+        // HTTP patch request.
+        private readonly HttpMethod _patch = new HttpMethod("PATCH");
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected DigitalTwinsClient()
+        {
+        }
+
+        internal DigitalTwinsClient(string hostName, IotHubConnectionProperties credentialProvider, HttpClient httpClient, HttpRequestMessageFactory httpRequestMessageFactory)
+        {
+            _hostName = hostName;
+            _credentialProvider = credentialProvider;
+            _httpClient = httpClient;
+            _httpRequestMessageFactory = httpRequestMessageFactory;
+        }
+
+        /// <summary>
+        /// Gets a strongly-typed digital twin.
+        /// </summary>
+        /// <param name="digitalTwinId">The Id of the digital twin.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The deserialized application/json digital twin and the ETag for the digital twin.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when the provided digital twin id is null.</exception>
+        /// <exception cref="ArgumentException">Thrown when the provided digital twin id is empty or whitespace.</exception>
+        /// <exception cref="IotHubException">
+        /// Thrown if IoT hub responded to the request with a non-successful status code. For example, if the provided
+        /// request was throttled, <see cref="IotHubThrottledException"/> is thrown. For a complete list of possible
+        /// error cases, see <see cref="Common.Exceptions"/>.
+        /// </exception>
+        /// <exception cref="HttpRequestException">
+        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
+        /// certificate validation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided cancellation token has requested cancellation.</exception>
+        public virtual async Task<GetDigitalTwinResponse<T>> GetAsync<T>(string digitalTwinId, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Getting digital twin with Id: {digitalTwinId}", nameof(GetAsync));
+
+            try
+            {
+                Argument.RequireNotNullOrEmpty(digitalTwinId, nameof(digitalTwinId));
+
+                cancellationToken.ThrowIfCancellationRequested();
+
+                using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(HttpMethod.Get, GetDigitalTwinRequestUri(digitalTwinId), _credentialProvider);
+                HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                await HttpMessageHelper2.ValidateHttpResponseStatus(HttpStatusCode.OK, response).ConfigureAwait(false);
```

Comment:
Q - why is this response code validator an async call? Don't we have the I/O operation result already, or are we making another call?

Reply:
The pre-existing exception mapping logic has a single async step in it for reading the contents of the http response. See [here](https://github.com/Azure/azure-iot-sdk-csharp/blob/be46adbbe5a0a82be8b84ead85d13e1c947ef1d3/iothub/service/src/Http2/HttpMessageHelper2.cs#L51)